### PR TITLE
feat(plugin) MutualGroupDMs: Add mutual groups to dm sidebar

### PR DESCRIPTION
**Pull request by Masterjoona**
https://github.com/Vencord/plugin-requests/issues/1046

### CODE REVIEW

File: src/plugins/mutualGroupDMs/index.tsx [hunk 1]
```diff
@@ -124,5 +133,35 @@ export default definePlugin({
...
+                                src={IconUtils.getChannelIconURL({ id: c.id, icon: c.icon, size: 32 })}
+                                size="SIZE_40"
+                                className={ProfileListClasses.listAvatar}
+                            >
+                            </Avatar>
+                            <div className={ProfileListClasses.listRowContent}>
+                                <div className={ProfileListClasses.listName}>{getGroupDMName(c)}</div>
+                                <div className={GuildLabelClasses.guildNick}>{c.recipients.length + 1} Members</div>
+                            </div>
+                        </Clickable>
```
Comment (Nuckyz) [inline src/plugins/mutualGroupDMs/index.tsx:161]: Extract this to a separate component so you can re-use in the other patch too

File: src/plugins/mutualGroupDMs/index.tsx [hunk 2]
```diff
@@ -70,6 +72,13 @@ export default definePlugin({
                     replace: "$1==='MUTUAL_GDMS'?$self.renderMutualGDMs(arguments[0]):$&"
                 }
             ]
+        },
+        {
+            find: 'section:"MUTUAL_FRIENDS"',
+            replacement: {
+                match: /,\i\)}\)}\)(?<=(\(0,\i\.jsx\)\(\i\.\i,{className:\i\.divider}\)).+?)/,
```
Comment (Nuckyz) [inline src/plugins/mutualGroupDMs/index.tsx:79]:
```suggestion
                match: /\.openUserProfileModal.+?\)}\)}\)(?<=(\(0,\i\.jsxs?\)\(\i\.\i,{className:\i\.divider}\)).+?)/,
```

File: src/plugins/mutualGroupDMs/index.tsx [hunk 3]
```diff
@@ -124,5 +133,35 @@ export default definePlugin({
                 }
             </ScrollerThin>
         );
+    }),
+
+    renderDMPageList: ErrorBoundary.wrap(({ user, Divider }: { user: User, Divider: JSX.Element; }) => {
+        const mutualGDms = getMutualGroupDms(user.id);
+        if (mutualGDms.length === 0) return null;
+        const header = getMutualGDMCountText(user);
+        return (
```
Comment (Nuckyz) [inline src/plugins/mutualGroupDMs/index.tsx:142]:
```suggestion
        const mutualGDms = getMutualGroupDms(user.id);
        if (mutualGDms.length === 0) return null;
        
        const header = getMutualGDMCountText(user);
        
        return (
```

File: src/plugins/mutualGroupDMs/index.tsx [hunk 4]
```diff
@@ -20,14 +20,16 @@ import ErrorBoundary from "@components/ErrorBoundary";
...
-import { findByPropsLazy } from "@webpack";
+import { findByPropsLazy, findComponentByCodeLazy } from "@webpack";
 import { Avatar, ChannelStore, Clickable, IconUtils, RelationshipStore, ScrollerThin, useMemo, UserStore } from "@webpack/common";
 import { Channel, User } from "discord-types/general";
 
 const SelectedChannelActionCreators = findByPropsLazy("selectPrivateChannel");
 const UserUtils = findByPropsLazy("getGlobalName");
 
 const ProfileListClasses = findByPropsLazy("emptyIconFriends", "emptyIconGuilds");
+const ListStyles = findByPropsLazy("list", "divider");
```
Comment (Nuckyz) [inline src/plugins/mutualGroupDMs/index.tsx:31]: Duplicate find, if you want you can grab the class from the patch

**Review by Nuckyz — CHANGES_REQUESTED** [review]

**Comment by Masterjoona** [comment]
Ty nuckyz, fixed

**Comment by Nuckyz** [comment]
Thank you for the addition!

**Pull request merged**